frontend/app/src/util/AppNavigation.ts

```diff
@@ -288,7 +288,13 @@ export class StrategyV2 {
         // The page name is embedded at the end of the URL path, and if not, we are in the main page.
         // See https://github.com/streamlit/streamlit/blob/1.19.0/frontend/src/App.tsx#L740
         pathname.endsWith("/" + appPage.urlPathname)
-      ) ?? this.mainPage
+      ) ??
+      this.mainPage ?? {
+        // A navigation call should always be called before this, but we want to provide something
```

Comment:
what do we provide here / what's the effect of returning an object with empty values?

Reply:
I'll actually look into removing it. I added it originally because I wanted to resolve some JS tests, but the truth is we expect this to not be null and we should adjust the test to ensure the proper flow. To be more specific, I think this occurs in the `App.tsx` tests because we are now ensuring a `NAVIGATION` proto message is being sent which would update these tests.

Reply:
Removing it is not an issue for now, so I will remove it for now.